File: python/api_key_gen.py
import secrets
import hashlib

class ApiKeyGen:
    def __init__(self, prefix="sk_live_"):
        self.prefix = prefix

    def generate(self):
        random_bytes = secrets.token_urlsafe(32)
        raw_key = f"{self.prefix}{random_bytes}"
        key_hash = hashlib.sha256(raw_key.encode()).hexdigest()
        return {"raw_key": raw_key, "db_hash": key_hash}
      

File: node/ApiKeyGen.js
const crypto = require('crypto');

class ApiKeyGen {
    constructor(prefix = 'sk_live_') {
        this.prefix = prefix;
    }

    generate() {
        const randomBytes = crypto.randomBytes(32).toString('base64url');
        const rawKey = `${this.prefix}${randomBytes}`;
        const dbHash = crypto.createHash('sha256').update(rawKey).digest('hex');
        return { rawKey, dbHash };
    }
}

module.exports = ApiKeyGen; // Essential for importing
